src/python/pants/engine/internals/graph_test.py

```diff
@@ -1678,9 +1756,8 @@ def test_dependency_inference(dependencies_rule_runner: RuleRunner) -> None:
     )
 
 
-def test_depends_on_subtargets(dependencies_rule_runner: RuleRunner) -> None:
-    """If the address is a BUILD target, or none of the dependency inference rules can infer
-    dependencies on sibling files, then we should depend on all the BUILD target's files."""
```

Comment:
> one of the dependency inference rules can infer dependencies on sibling files

That will be removed in a followup - the target generator rules should do this, not the dependency rule.